config/application.rb

```diff
@@ -117,6 +117,9 @@ def config.database_configuration
 
     Rails.autoloaders.main.ignore(Dir["#{config.root}/app/models/reports"])
 
+    require_dependency 'lib/zeitwerk/zeitwerk.rb'
```

Comment:
Just curious, why do we need to manually require zeitwerk here?

Reply:
That was my mistake that I put that file inside freedom_patches because it is more like a config. It contains a map for Zeitwerk for classes named incorrectly, at least based on filename convention. Stuff like:

```
'topic_query_sql' => 'TopicQuerySQL',
'clean_up_unmatched_ips' => 'CleanUpUnmatchedIPs',
'pop3_polling_enabled_setting_validator' => 'POP3PollingEnabledSettingValidator',
```

It worked because initializers/002-freedom-patches.rb was evaluated before Zeitwerk looked for those classes and was able to map them correctly.

Since we moved freedom patches do after initialize block (Rails.application.reloader.to_prepare do) that map is loaded too late. 

I decided to move that file to `lib/zeitwerk` and require it by hand

Reply:
ahhh icic :+1: